contracts/SimpleSwap.sol

```diff
@@ -116,53 +97,39 @@ contract SimpleSwap is SW3Utils {
   /// @param value maximum amount to send
   /// @return payout amount that was actually paid out
   /// @return payout amount that bounced
-  function _payout(address beneficiary, uint value) internal returns (uint payout, uint bounced) {
-    /* part of hard deposit used */
-    payout = Math.min256(value, hardDeposits[beneficiary].amount);
-    /* if there some of the hard deposit is used update the structure */
-    if(payout != 0) {
-      hardDeposits[beneficiary].amount -= payout;
-      totalDeposit -= payout;
+  function _payout(address beneficiary, uint amount) internal returns (bool) {
+    /* SWAP contract should be allowed to process the payout */
+    if(amount >= liquidBalance()) {
+      return false; // returning false signals a bounced cheque. TODO: consider reverting
     }
 
-    /* amount of the cash not backed by a hard deposit */
-    uint rest = value - payout;
-    uint liquid = liquidBalance();
-
-    if(liquid >= rest) {
-      /* swap channel is solvent */
-      payout = value;
-    } else {
-      /* part of the cheque bounces */
-      payout += liquid;
-      bounced = rest - liquid;
-    }
+    /* Update internal accounting */
+    hardDeposits[beneficiary].amount = hardDeposits[beneficiary].amount.sub(amount);
+    totalDeposit = totalDeposit.sub(amount);
 
     /* transfer the payout */
-    beneficiary.transfer(payout);
+    beneficiary.transfer(amount);
+    return true;
   }
 
   /// @notice attempt to cash latest cheque
   /// @param beneficiary beneficiary for whose cheque should be paid out
-  function cashCheque(address beneficiary) public {
+  function cashCheque(address beneficiary, uint256 amount) public {
     ChequeInfo storage info = cheques[beneficiary];
 
     /* grace period must have ended */
     require(now >= info.timeout);
 
     /* ensure there is actually ether to be paid out */
-    uint value = info.amount.sub(info.paidOut); /* throws if paidOut > amount */
-    require(value > 0);
-
-    /* do the actual payout */
-    (uint payout, uint bounced) = _payout(beneficiary, value);
+    require(info.amount.sub(info.paidOut) >= amount);
+    info.paidOut = info.paidOut.add(amount);
```

Comment:
I do the paidOut accounting before actually attempting to do a transfer (handing over control flow to another contract/account).